test/functional/test_framework/util.py

```diff
@@ -28,6 +28,10 @@
 
 def assert_approx(v, vexp, vspan=0.00001):
     """Assert that `v` is within `vspan` of `vexp`"""
+    if isinstance(v, Decimal) or isinstance(vexp, Decimal):
+        v=Decimal(v)
+        vexp=Decimal(vexp)
+        vspan=Decimal(vspan)
```

Comment:
This results in `Decimal('0.000010000000000000000818030539140313095458623138256371021270751953125')`

Better to just insist `v` and `vexp` have compatible types, and default `vspan` sanely?

Reply:
The idea is to let the caller not worry too much about converting stuff to/from Decimal instances. Is there an issue resulting in `vspan` having some noise somewhere?